src/browser/provider/built-in.js

```diff
@@ -0,0 +1,62 @@
+import browserNatives from 'testcafe-browser-natives';
+import { assignIn } from 'lodash';
+
+
+export var PathBrowserProvider = {
+    async openBrowser (id, descriptor, startPage) {
+        await browserNatives.open(descriptor, startPage);
+    },
+
+    async closeBrowser (id, pageInfo) {
+        await browserNatives.close(pageInfo.title);
+    },
+
+    async getDescriptorDefaultValues (alias) {
+        return browserNatives.getBrowserInfo(alias);
+    },
+
+    async resizeWindow (id, pageInfo, width, height) {
+        await browserNatives.resize(pageInfo.title, pageInfo.width, pageInfo.height, width, height);
+
+        /*eslint-disable prefer-arrow-callback*/
+        /*eslint-disable no-undef*/
+        var { newWidth, newHeight } = await this.evalOnClient(id, function () {
+            return { newWidth: window.innerWidth, newHeight: window.innerHeight };
+        });
+        /*eslint-enable no-undef*/
+        /*eslint-enable prefer-arrow-callback*/
+
+        if (newWidth !== width || newHeight !== height)
+            await browserNatives.resize(pageInfo.title, newWidth, newHeight, width, height);
+    },
+
+    async screenshotWindow (id, pageInfo, screenshotPath) {
+        await browserNatives.screenshot(pageInfo.title, screenshotPath);
+    }
+};
+
+export var LocalBrowserProvider = assignIn({}, PathBrowserProvider, {
+    async listAvailableAliases () {
+        return Object.keys(await browserNatives.getInstallations());
+    },
+
+    async isValidAlias (alias) {
+        return (await this.listAvailableAliases()).indexOf(alias) !== -1;
+    }
+});
+
+export var RemoteBrowserProvider = {
+    // NOTE: we do not have control over remote browser
+    async openBrowser () {
+        return;
+    },
+
+    async closeBrowser () {
+        return;
+    },
+
+    // NOTE: we must try to do a local screenshot or resize
+    screenshotWindow: LocalBrowserProvider.screenshotWindow,
+
+    resizeWindow: LocalBrowserProvider.resizeWindow
```

Comment:
BTW, since we now have `providername:descr` notation, maybe we should use `remote:1`, `remote:2` and so on?